core/tests/protractor_utils/ExplorationPlayerPage.js

```diff
@@ -0,0 +1,164 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Page object for the exploration player, for use in Protractor
+ * tests.
+ */
+
+var forms = require('./forms.js');
+var general = require('./general.js');
+var interactions = require('../../../extensions/interactions/protractor.js');
+
+var ExplorationPlayerPage = function() {
+  var conversationInputAll =
```

Comment:
Consider arranging all these elements so that you have non-interactive components first, then a blank line, then buttons, then a blank line, then links. It's a bit easier to understand this long list if we group them by function, I think.

Reply:
Ok, ya, will do that.

Reply:
Done